lib/publishing.ts

```diff
@@ -113,6 +118,10 @@ export class AssetPublishing implements IPublishProgress {
       },
       dockerFactory: new DockerFactory(),
     };
+    if (options.progressListener) {
+      globalOutputHandler.setListener(options.progressListener);
```

Comment:
So what happens if the CDK (or any consumer) instantiates two instances of `AssetPublishing` with different listeners? If this is not expected behavior - protect against it. If this is expected behavior, how would it work? 

Right now, this is a subtle bug waiting to happen.